clustpy/deep/_data_utils.py

```diff
@@ -224,6 +224,125 @@ def get_dataloader(X: np.ndarray | torch.Tensor, batch_size: int, shuffle: bool
+    batch_size : int
+        the batch size (default: 256)
+    custom_dataloaders : tuple
+        tuple consisting of a trainloader (random order) at the first and a test loader (non-random order) at the second position.
+        Can also be a tuple of strings, where the first entry is the path to a saved trainloader and the second entry the path to a saved testloader.
+        In this case the dataloaders will be loaded by torch.load(PATH).
+        If None, the default dataloaders will be used (default: None)
+
+    Returns
+    -------
+    tuple : (torch.utils.data.DataLoader, torch.utils.data.DataLoader, int)
+        The trainloader,
+        The testloader,
+        The batch size (can be different from input if another value is used within custom_dataloader)
+    """
+    # sample random mini-batches from the data -> shuffle = True
+    if custom_dataloaders is None:
+        trainloader = get_dataloader(X, batch_size, True, False)
+        testloader = get_dataloader(X, batch_size, False, False)
+    else:
+        trainloader, testloader = custom_dataloaders
+        # If train-/testloader is string, it can be loaded from a file
+        if type(trainloader) is str:
+            trainloader = torch.load(trainloader)
+        if type(testloader) is str:
+            testloader = torch.load(testloader)
+        if trainloader.batch_size != testloader.batch_size:
+            print(
+                "INFO: Batch size of trainloader and testloader do not match: trainloader = {0}, testloader = {1}".format(
+                    trainloader.batch_size, testloader.batch_size))
+        if trainloader.batch_size != batch_size:
+            print(
+                "WARNING: Specified batch_size differs from trainloader.batch_size. Will use trainloader.batch_size ({0}).".format(
+                    trainloader.batch_size))
+            batch_size = trainloader.batch_size
+    return trainloader, testloader, batch_size
+
+
+def get_default_augmented_dataloaders(X: np.ndarray | torch.Tensor, batch_size: int = 256, conv_used: bool = False,
+                                      flatten: bool = True) -> (
+        torch.utils.data.DataLoader, torch.utils.data.DataLoader):
+    """
+    Receive a train- and a test dataloader using default augmentations.
+    These transformations correspond to a min-max normalization followed by
+    torchvision.transforms.RandomAffine(degrees=(-16, +16), translate=(0.1, 0.1), shear=(-8, 8), fill=0) and
+    a channel-wise z-transformation.
+    Optionally, the images can be flatten afterward.
+
+
+    Parameters
+    ----------
+    X : np.ndarray | torch.Tensor
+        the actual data set (can be np.ndarray or torch.Tensor)
+    batch_size : int
+        the batch size (default: 256)
+    conv_used : bool
+        defines whether a convolutional network will be used afterward.
+        In this case, grayscale images will be transformed to receive three color channels (default: False)
```

Comment:
maybe change 
>  In this case, grayscale images will be transformed to receive three color channels (default: False)
to
 In this case, grayscale images will be transformed to receive three color channels by copying the grayscale channel three times (default: False)